db/helpers/generatorUtil.js

```diff
@@ -13,6 +13,41 @@ const common = require('./common');
 const dbErrors = require('../dbErrors');
 
 /**
+ * Function to validate the context field of the sample generator based on the
+ * contextDefinition field of its related generator template.
+ * @param  {Object} sgCtx - The generator context object     [description]
+ * @param  {Object} sgtCtxDef - The related sample generator template context
+ * definition object
+ * @throws {MissingRequiredFieldrror} If the generator context field does not
+ * have the attributes that are required by the context definiton field of the
+ * sample generator template
+ * @throws {ValidationError} If the number of keys in the generator context do
+ * not match the number of keys in the generator template context.
+ */
+function validateGeneratorCtx(sgCtx, sgtCtxDef) {
+  const sgtCtxDefKeys = sgtCtxDef ? Object.keys(sgtCtxDef) : [];
+  const sgCtxKeys = sgCtx ? Object.keys(sgCtx) : [];
+  const sgtCtxDefKeysSet = new Set(sgtCtxDefKeys);
```

Comment:
Converting to a set is interesting--makes it seem like it should *also* be an error if you have the same key defined more than once in ctx and/or ctx def, right?

Reply:
Created a user story for this. Looks like we have the same problem for all the JSON/JSONB columns. The story covers adding this validation across all the refocus models.